client/extensions/woocommerce/state/data-layer/products/index.js

```diff
@@ -75,3 +83,22 @@ export function handleProductRequest( { dispatch }, action ) {
 	const updatedSuccessAction = updatedAction( siteId, action, successAction );
 	dispatch( get( siteId, 'products/' + productId, updatedSuccessAction, failureAction ) );
 }
+
+export function handleProductsRequestSuccess( store, action, next, response ) {
+	const { body, status } = response.data;
+
+	// We needed to envelope the reponse to get the number of products and pages. see handleProductsRequest
+	if ( status !== 200 ) {
+		action.meta.dataLayer.error = { code: ( body.code || status ), message: body.message };
```

Comment:
Unfortunately, it looks like reaching into `action.meta` from the reducer is not recommended. That stance is taken here in another PR: https://github.com/Automattic/wp-calypso/pull/16385#issuecomment-317588993

I'll also be updating my existing product category PR to no longer use this functionality.